modules/search/instant-search/components/search-app.jsx

```diff
@@ -61,56 +60,91 @@ class SearchApp extends Component {
 		} );
 	}
 
+	componentDidMount() {
+		if ( this.props.grabFocus ) {
+			this.input.current.focus();
+		}
+	}
+
+	hasActiveQuery() {
+		return this.state.query !== '' || hasFilter();
+	}
+
+	activateResults() {
+		if ( ! this.state.resultsActive ) {
+			this.setState( { resultsActive: true } );
```

Comment:
[`setState` is not synchronous](https://reactjs.org/docs/react-component.html#setstate), so we'll want to do something like the following:
```js
this.setState( { resultsActive: true }, () => { doSomethingHere(); } );
```

We might also want `activateResults` to accept a `callback` input like so:
```js
activateResults( callback = () => {} ) {
```